dkg-gadget/src/worker.rs

```diff
@@ -921,42 +982,55 @@ where
 	///    method will try to restart the keygen protocol.
 	fn maybe_enact_next_authorities(&self, header: &B::Header) {
 		if !self.should_execute_new_keygen(header) {
-			debug!(target: "dkg_gadget::worker", "🕸️  Not executing new keygen protocol");
+			self.logger.debug("🕸️  Not executing new keygen protocol");
 			return
 		}
 
 		// Get the active and queued validators to check for updates
 		if let Some((_active, queued)) = self.validator_set(header) {
-			debug!(target: "dkg_gadget::worker", "🕸️  Session progress percentage above threshold, proceed with enact new authorities");
+			self.logger.debug("🕸️  Session progress percentage above threshold, proceed with enact new authorities");
 			// Check if there is a keygen is finished:
 			let queued_keygen_finished = self
 				.next_rounds
 				.read()
 				.as_ref()
 				.map(|r| r.is_keygen_finished())
 				.unwrap_or(false);
-			debug!(target: "dkg_gadget::worker", "🕸️  QUEUED KEYGEN FINISHED: {:?}", queued_keygen_finished);
-			debug!(target: "dkg_gadget::worker", "🕸️  QUEUED DKG STATUS: {:?}", self.next_rounds.read().as_ref().map(|r| r.status.clone()));
+			self.logger
+				.debug(format!("🕸️  QUEUED KEYGEN FINISHED: {queued_keygen_finished:?}"));
+			self.logger.debug(format!(
+				"🕸️  QUEUED DKG STATUS: {:?}",
+				self.next_rounds.read().as_ref().map(|r| r.status.clone())
+			));
 			if queued_keygen_finished {
 				return
 			}
 
 			let has_next_rounds = self.next_rounds.read().is_some();
-			debug!(target: "dkg_gadget::worker", "🕸️  HAS NEXT ROUND KEYGEN: {:?}", has_next_rounds);
+			self.logger.debug(format!("🕸️  HAS NEXT ROUND KEYGEN: {has_next_rounds:?}"));
 			// Check if there is a next DKG Key on-chain.
 			let next_dkg_key = self.get_next_dkg_pub_key(header);
 
-			debug!(target: "dkg_gadget::worker", "🕸️  NEXT DKG KEY ON CHAIN: {}", next_dkg_key.is_some());
+			self.logger
+				.debug(format!("🕸️  NEXT DKG KEY ON CHAIN: {}", next_dkg_key.is_some()));
+			let test_harness_mode = self.to_test_client.is_some();
 			// Start a keygen if we don't have one OR if there is no queued key on chain.
-			if !has_next_rounds && next_dkg_key.is_none() {
-				debug!(target: "dkg_gadget::worker", "🕸️  NO NEXT ROUND KEYGEN AND NO NEXT DKG | STARTING A NEW QUEUED DKG: {}", next_dkg_key.is_some());
+			if (!has_next_rounds && next_dkg_key.is_none()) || test_harness_mode {
```

Comment:
So we always execute a new keygen for the test harness, is this used to push time forward effectively?